tiledb/highlevel.py

```diff
@@ -164,19 +163,118 @@ def delete_fragments(
 
     vfs = tiledb.VFS(config=config, ctx=ctx)
 
+    if verbose or dry_run:
+        print("Deleting fragments:")
+
     for frag in tiledb.array_fragments(uri):
         if (
             timestamp_range[0] <= frag.timestamp_range[0]
             and frag.timestamp_range[1] <= timestamp_range[1]
         ):
             if verbose or dry_run:
-                print(frag.uri)
+                print(f"\t{frag.uri}")
 
             if not dry_run:
                 vfs.remove_file(f"{frag.uri}.ok")
                 vfs.remove_dir(frag.uri)
 
 
+def create_array_from_fragments(
+    src_uri,
+    dst_uri,
+    timestamp_range,
+    config=None,
+    ctx=None,
+    verbose=False,
+    dry_run=False,
+):
+    """
+    (POSIX only). Create a new array from an already existing array by selecting
+    fragments that fall withing a given timestamp_range. The original array is located
+    at src_uri and the new array is created at dst_uri.
+
+    :param str src_uri: URI for the source TileDB array (any supported TileDB URI)
+    :param str dst_uri: URI for the newly created TileDB array (any supported TileDB URI)
+    :param (int, int) timestamp_range: (default None) If not None, vacuum the
+        array using the given range (inclusive)
+    :param config: Override the context configuration. Defaults to ctx.config()
+    :param ctx: (optional) TileDB Ctx
+    :param verbose: (optional) Print fragments being copied (default: False)
+    :param dry_run: (optional) Preview fragments to be copied without
+        running (default: False)
+    """
+    if not isinstance(timestamp_range, tuple) and len(timestamp_range) != 2:
+        raise TypeError(
+            "'timestamp_range' argument expects tuple(start: int, end: int)"
+        )
+
+    if not ctx:
+        ctx = tiledb.default_ctx()
+
+    if config is None:
+        config = tiledb.Config(ctx.config())
+
+    vfs = tiledb.VFS(config=config, ctx=ctx)
+
+    fragment_info = tiledb.array_fragments(src_uri)
+
+    if len(fragment_info) < 1:
+        print("Cannot create new array; no fragments to copy")
+        return
+
+    if verbose or dry_run:
+        print(f"Creating directory for array at {dst_uri}\n")
+
+    if not dry_run:
+        vfs.create_dir(dst_uri)
+
+    src_lock = os.path.join(src_uri, "__lock.tdb")
+    dst_lock = os.path.join(dst_uri, "__lock.tdb")
+
+    if verbose or dry_run:
+        print(f"Copying lock file {dst_uri}\n")
+
+    if not dry_run:
+        vfs.copy_file(f"{src_lock}", f"{dst_lock}")
+
+    for frag in fragment_info:
+        if not (
+            timestamp_range[0] <= frag.timestamp_range[0]
+            and frag.timestamp_range[1] <= timestamp_range[1]
+        ):
+            continue
+
+        schema_name = fragment_info.array_schema_name[0]
```

Comment:
Need to use `frag.array_schema_name`. Currently this is only copying the first array schema version, so the array is corrupted after copy.